### PR TITLE
Add required-without-page_token information to pagination extension

### PR DESCRIPTION
**Pull request by ahl**
Previously `x-dropshot-pagination` was simply the value `true`. Now it's an object with a field `required` that is an array of strings that tell us what query parameters are required to fetch the first page (i.e. when `page_token` is not specified).

This also cleans up some stray doc nits.

### CODE REVIEW

File: dropshot/src/lib.rs [hunk 1]
```diff
@@ -507,15 +507,35 @@
 //! this ought to work, but it currently doesn't due to serde_urlencoded#33,
 //! which is really serde#1183.
 //!
-//! ### DTrace probes
+//! Note that any parameters defined by `MyScanParams` are effectively encoded
+//! into the page token and need not be supplied invocations when `page_token`
```
Comment (davepacheco) [inline dropshot/src/lib.rs:511]:
```suggestion
//! into the page token and need not be supplied with invocations when `page_token`
```

File: dropshot/src/lib.rs [hunk 2]
```diff
@@ -507,15 +507,35 @@
...
+//! is specified. That is not the case for required parameters defined by
+//! `MyExtraQueryParams`--those must be supplied on each invocation.
+//!
+//! ### OpenAPI extension
+//!
+//! In generated OpenAPI documents, Dropshot adds the `x-dropshot-paginated`
+//! extension to paginated operations. The value is currently a structure
+//! with this format:
+//! ```json
```
Comment (davepacheco) [inline dropshot/src/lib.rs:520]: Does this render right without a newline here?

**Review by davepacheco — APPROVED** [review]
Do I grok correctly that there's no impact on dropshot consumers here (unless they're using the openapi extension)?

File: dropshot/src/lib.rs (hunk 2 above)
Comment (ahl) [reply dropshot/src/lib.rs:520]: image

Yes it looks good to me.

**Comment by ahl** [comment]
> Do I grok correctly that there's no impact on dropshot consumers here (unless they're using the openapi extension)?

Yes; that's accurate.

**Pull request merged**